Possible panic on transport re-establishment.

```
[2019-06-13T18:00:11+08:00] INFO [trmanager]: Failed to re-dial Transport f531c4a4-013d-59ea-866d-03747b51b914: deadline exceeded
[2019-06-13T18:00:11+08:00] INFO [trmanager]: Transport f531c4a4-013d-59ea-866d-03747b51b914 failed with error: EOF. Re-dialing...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x15e0a14]

goroutine 91 [running]:
github.com/skycoin/skywire/pkg/transport.(*ManagedTransport).Close(0x0, 0xc000252240, 0xc000505bb0)
        /Users/evanlinjin/GolandProjects/skywire/pkg/transport/managed_transport.go:81 +0x34
github.com/skycoin/skywire/pkg/transport.(*Manager).DeleteTransport(0xc0000c0780, 0xea593d01a4c431f5, 0x14b9517b74036d86, 0xc00013f700, 0x2)
        /Users/evanlinjin/GolandProjects/skywire/pkg/transport/manager.go:216 +0x142
github.com/skycoin/skywire/pkg/transport.(*Manager).manageTransport(0xc0000c0780, 0x19cd3e0, 0xc0000c8020, 0xc00012a080, 0x19ce820, 0xc000212460, 0x55a01c541c94803, 0x7fe5f28a23ff55c4, 0x4b6aa99c314f9b8, 0xc1a4912bb3b5abe9, ...)
        /Users/evanlinjin/GolandProjects/skywire/pkg/transport/manager.go:386 +0xd7c
created by github.com/skycoin/skywire/pkg/transport.(*Manager).acceptTransport
        /Users/evanlinjin/GolandProjects/skywire/pkg/transport/manager.go:343 +0xadd
```

--- a/pkg/transport/manager.go
+++ b/pkg/transport/manager.go
@@ -3,6 +3,7 @@ package transport
 import (
 	"context"
 	"errors"
+	"io"
 	"math/big"
 	"strings"
 	"sync"
@@ -241,21 +242,17 @@ func (tm *Manager) CreateTransport(ctx context.Context, remote cipher.PubKey, tp
 // DeleteTransport disconnects and removes the Transport of Transport ID.
 func (tm *Manager) DeleteTransport(id uuid.UUID) error {
 	tm.mu.Lock()
-	tr := tm.transports[id]
-	delete(tm.transports, id)
+	if tr, ok := tm.transports[id]; ok {
+		delete(tm.transports, id)
+		_ = tr.Close() //nolint:errcheck
+	}
 	tm.mu.Unlock()
-
-	tr.Close()
 
 	if _, err := tm.config.DiscoveryClient.UpdateStatuses(context.Background(), &Status{ID: id, IsUp: false}); err != nil {
 		tm.Logger.Warnf("Failed to change transport status: %s", err)
 	}
 
 	tm.Logger.Infof("Unregistered transport %s", id)
-	if tr != nil {
-		return tr.Close()
-	}
-
 	return nil
 }
 
@@ -330,11 +327,13 @@ func (tm *Manager) createTransport(ctx context.Context, remote cipher.PubKey, tp
 	tm.transports[entry.ID] = mTr
 	tm.mu.Unlock()
 
-	tm.TrChan <- mTr
-
-	go tm.manageTransport(ctx, mTr, factory, remote, public, false)
-
-	return mTr, nil
+	select {
+	case <-tm.doneChan:
+		return nil, io.ErrClosedPipe
+	case tm.TrChan <- mTr:
+		go tm.manageTransport(ctx, mTr, factory, remote, public, false)
+		return mTr, nil
+	}
 }
 
 func (tm *Manager) acceptTransport(ctx context.Context, factory Factory) (*ManagedTransport, error) {
@@ -371,11 +370,13 @@ func (tm *Manager) acceptTransport(ctx context.Context, factory Factory) (*Manag
 	tm.transports[entry.ID] = mTr
 	tm.mu.Unlock()
 
-	tm.TrChan <- mTr
-
-	go tm.manageTransport(ctx, mTr, factory, remote, true, true)
-
-	return mTr, nil
+	select {
+	case <-tm.doneChan:
+		return nil, io.ErrClosedPipe
+	case tm.TrChan <- mTr:
+		go tm.manageTransport(ctx, mTr, factory, remote, true, true)
+		return mTr, nil
+	}
 }
 
 func (tm *Manager) walkEntries(walkFunc func(*Entry) bool) *Entry {

